src/PhoneHighlightYS.cpp

```diff
@@ -0,0 +1,109 @@
+// Copyright [2019] <Stanislavyk Yaroslav> (stl.ros@outlook.com)
+
+#include "PhoneHighlightYS.h"
+
+#include <cstdint>
+#include <tuple>
+
+namespace {
+
+constexpr uint8_t kPhoneNumber = 0u;
+constexpr uint8_t kSearchString = 1u;
+constexpr uint8_t kSearchStringPos = 2u;
+constexpr uint8_t kMatchingStatus = 3u;
+constexpr uint8_t kRange = 4u;
+constexpr uint8_t kHighlightRanges = 5u;
+
+template <typename Args>
+void CollectRange(Args const& args)
+{
+    std::get<kMatchingStatus>(args) = false;
+    std::get<kHighlightRanges>(args).push_back(std::get<kRange>(args));
+}
+
+template <typename Symbol, typename IgnoredSymbols>
```

Comment:
@Morheit , why you need template?

Reply:
@malirod, it was handy during the implementation. All templates have been removed